lucene/core/src/java/org/apache/lucene/store/DataInput.java

```diff
@@ -92,15 +92,20 @@ public void readBytes(byte[] b, int offset, int len, boolean useBuffer)
    * @see DataOutput#writeByte(byte)
    */
   public short readShort() throws IOException {
-    return (short) (((readByte() & 0xFF) <<  8) |  (readByte() & 0xFF));
+    final byte b1 = readByte();
+    final byte b2 = readByte();
+    return (short) ((b2 << 8) | (b1 & 0xFF));
```

Comment:
can you keep the conversion to an unsigned int of b2 like we had before?

Reply:
done